search/url.go

```
@@ -0,0 +1,58 @@
+package search
+
+import (
+	"net/url"
+
+	"github.com/PuerkitoBio/purell"
+)
+
+var trackingParams = []string{
+	"__s",
+	"_ga",
+	"campaign_id",
+	"ceid",
+	"emci",
+	"emdi",
+	"fbclid",
+	"gclid",
+	"hootPostID",
+	"mc_eid",
+	"mkclid",
+	"mkt_tok",
+	"msclkid",
+	"pk_campaign",
+	"pk_kwd",
+	"sessionid",
+	"sourceid",
+	"utm_campaign",
+	"utm_content",
+	"utm_id",
+	"utm_medium",
+	"utm_source",
+	"utm_term",
+	"xpid",
+}
+
+// aggressively normalizes URL, for search indexing and matching. it is possible the URL won't be directly functional after this normalization
+func NormalizeLossyURL(raw string) string {
+	clean, err := purell.NormalizeURLString(raw, purell.FlagsUsuallySafeGreedy|purell.FlagRemoveDirectoryIndex|purell.FlagRemoveFragment|purell.FlagRemoveDuplicateSlashes|purell.FlagRemoveWWW|purell.FlagSortQuery)
+	if err != nil {
+		return raw
+	}
+
+	// remove tracking params
+	u, err := url.Parse(clean)
```

Think we need some error handling here. `u` can be nil and we ignore this error